indexers/staking-squid/src/events/bundle.ts

```diff
@@ -58,50 +59,61 @@ export function processBundleStoredEvent(
-    totalVolume,
-    consensusStorageFee: BigInt(blockFees.consensusStorageFee),
-    domainExecutionFee: BigInt(blockFees.domainExecutionFee),
-    burnedBalance: BigInt(blockFees.burnedBalance),
-  });
-
-  cache.bundles.set(bundle.id, bundle);
-
-  domain.lastDomainBlockNumber = Number(domainBlockNumber);
-  domain.totalTransfersIn += totalTransfersIn;
-  domain.transfersInCount += transfersInCount;
-  domain.totalTransfersOut += totalTransfersOut;
-  domain.transfersOutCount += transfersOutCount;
-  domain.totalRejectedTransfersClaimed += totalRejectedTransfersClaimed;
-  domain.rejectedTransfersClaimedCount += rejectedTransfersClaimedCount;
-  domain.totalTransfersRejected += totalTransfersRejected;
-  domain.transfersRejectedCount += transfersRejectedCount;
-  domain.totalVolume += totalVolume;
-  domain.totalConsensusStorageFee += BigInt(blockFees.consensusStorageFee);
-  domain.totalDomainExecutionFee += BigInt(blockFees.domainExecutionFee);
-  domain.totalBurnedBalance += BigInt(blockFees.burnedBalance);
-  domain.bundleCount++;
-  domain.lastBundleAt = getBlockNumber(block);
-  domain.updatedAt = getBlockNumber(block);
-
-  cache.domains.set(domain.id, domain);
-
+  let bundle = cache.bundles.get(bundleUID(domainId, domainBlockHash));
+
+  if (!bundle) {
+    bundle = createBundle(domain.id, domainBlockHash, {
+      domainBlockNumber: Number(domainBlockNumber),
+      domainBlockHash,
+      domainBlockExtrinsicRoot,
+      consensusBlockNumber: Number(consensusBlockNumber),
+      consensusBlockHash,
+      totalTransfersIn,
+      transfersInCount,
+      totalTransfersOut,
+      transfersOutCount,
+      totalRejectedTransfersClaimed,
+      rejectedTransfersClaimedCount,
+      totalTransfersRejected,
+      transfersRejectedCount,
+      totalVolume,
+      consensusStorageFee: BigInt(blockFees.consensusStorageFee),
+      domainExecutionFee: BigInt(blockFees.domainExecutionFee),
+      burnedBalance: BigInt(blockFees.burnedBalance),
+    });
+    cache.bundles.set(bundle.id, bundle);
+
+    const bundleAuthor = createBundleAuthor(bundle.id, account.id, operator.id);
+    cache.bundleAuthors.set(bundleAuthor.id, bundleAuthor);
+
+    domain.lastDomainBlockNumber = Number(domainBlockNumber);
+    domain.totalTransfersIn += totalTransfersIn;
+    domain.transfersInCount += transfersInCount;
+    domain.totalTransfersOut += totalTransfersOut;
+    domain.transfersOutCount += transfersOutCount;
+    domain.totalRejectedTransfersClaimed += totalRejectedTransfersClaimed;
+    domain.rejectedTransfersClaimedCount += rejectedTransfersClaimedCount;
+    domain.totalTransfersRejected += totalTransfersRejected;
+    domain.transfersRejectedCount += transfersRejectedCount;
+    domain.totalVolume += totalVolume;
+    domain.totalConsensusStorageFee += BigInt(blockFees.consensusStorageFee);
+    domain.totalDomainExecutionFee += BigInt(blockFees.domainExecutionFee);
+    domain.totalBurnedBalance += BigInt(blockFees.burnedBalance);
```

Comment:
Are these going to double count values from other bundles with the same execution receipt?

Reply:
So the v11 has a bundle and a bundle_author table, if the bundle already exist we don't addition these again we just add the new author (operator)

Now the part that I have currently intentionally skip, is trying to weight somehow the addition made to the operator to correspond to his % of the domain stake

Reply:
> Now the part that I have currently intentionally skip, is trying to weight somehow the addition made to the operator to correspond to his % of the domain stake

Any ideas on how to handle this yet?

Reply:
I'm not sure no, and I'm also not sure if this is even a problem that need to be changed.

We are not trying to calculate the reward accumulated by the operator here, but capturing what activity the operator has been part of.

For example, when XDM start get use, these stats will capture transfer and rejected transfer. This will then reveil if some operators reject transfer or process transfer more than others, for example, so then starting to weight these total will make little sense...

Obviously, I'm open to suggestion, otherwise I'll suggest we gather feedback ounce it's live in prod and especially when xdm start to be used, to see what changes in these kind of stats should be made

Reply:
Great point and I agree, it doesn't matter for operators if activity is double counted.